.github/workflows/cibuild.yml

```diff
@@ -22,16 +22,17 @@ jobs:
         os: [ ubuntu-latest, windows-latest ]
     steps:
       # Starting in v2.2 checkout action fetches all tags when fetch-depth=0, for auto-versioning.
-      - uses: actions/checkout@v2.3.2
+      - uses: actions/checkout@v3.1.0
         with:
           fetch-depth: 0
       # Java setup step completes very fast, no need to run in a preconfigured docker container
       - name: Set up JDK 17
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: 17
+          distribution: adopt
```

Comment:
```suggestion
          distribution: temurin
```

Adopt OpenJDK is deprecated

Also this seems to have [support for caching](https://github.com/actions/setup-java/pull/193), which we should probably instead of manually configuring `actions/cache`